### PR TITLE
use Unidecode

### PR DESCRIPTION
**Pull request by chylli**

### CODE REVIEW

File: share/romanization.yml [hunk 1]
```diff
@@ -1,82 +1,152 @@
...
-υ: y
-υι: yi
-φ: f
-ϕ: f
-χ: ch
-ψ: ps
-ω: o
-ωυ: oy
-
+# https://docs.google.com/document/d/1L_zN6JS5hANo7Q31ros1xoZajPou5T4MIUo_06-dTDE/edit
```
Comment (basiliscos) [inline share/romanization.yml]: I'm not sure that referring our private document in public module is good idea. If some document is available on `https://www.mfsa.com.mt/`  better to refer it.

Comment (chylli) [reply share/romanization.yml]: You are right. I have removed this line.

**Comment by raunakkathuria** [comment]
can you check failing tests

File: lib/Finance/MIFIR/CONCAT.pm [hunk 2]
```diff
@@ -58,10 +56,8 @@ sub mifir_concat {
 sub _process_name {
     my ($str) = @_;
     $str = lc($str);
-    $str =~ s/$_/$romanization->{$_}/g for keys %$romanization;
     $str =~ s/$_\s+//g for (@{$config->{titles}}, @{$config->{prefixes}});
-    $str =~ s/’//g;    # our iconv does not handle this correctly, it returns empty string if we have it
-    $str = $converter->convert($str);
+    $str =~ s/$_/$romanization->{$_}/g for keys %$romanization;
```
Comment (torsten-deriv) [inline lib/Finance/MIFIR/CONCAT.pm]: The `%romanization` hash is pretty long and you are compiling and running a separate regex for each element. Why don't you combine them in one regex and compile it once. Something along these lines:

```perl
my $re=join '', keys %$romanization;
$re=qr/([$re])/;            # provided the keys are not any special characters like \, - or ]

sub _process_name {
...
    $str =~ s/$re/$romanization->{$1}/ge;
}
```

Depending on how many titles and prefixes there are that might also be a better approach for line 59.

File: t/mifir.t [hunk 3]
```diff
@@ -32,11 +32,38 @@ is mifir_concat({
...
+is Finance::MIFIR::CONCAT::_process_name('íśÏșÛ'), 'isisu', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('đûËŵĈ'), 'duewc', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('ùüžĴð'), 'uuzjd', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('ŜŠÈŔã'), 'ssera', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('ĆŶäȘũ'), 'cyasu', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('ÚŞǎöĂ'), 'usaoa', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('þŻÓøẞ'), 'tzoos', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('ôļŘķň'), 'olrkn', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('źĶŷÉń'), 'zkyen', 'character check';
+is Finance::MIFIR::CONCAT::_process_name('ÌÞŸĜÑ'), 'itygn', 'character check';
```
Comment (zakame) [inline t/mifir.t]: Hi, just commenting on Perl style here (as well as getting used to the codebase in general.)

Are the character check tests exhaustive and call all combinations of these characters?  If not, would it make sense to generate all these combinations and put into another file that this test script can loop over?

Comment (chylli) [reply t/mifir.t]: Hi @zakame , I don't think it is necessary to test all combinations here. It is enough to make sure all characters in transliteration tables are replaced with the right character. We don't care about combinations. But I extract all strings into a variable now.

Comment (zakame) [reply t/mifir.t]: Cool, no worries :+1:

**Pull request merged**